From 48047b9696e3a63a015d98044ddcdd76efbd3c2e Mon Sep 17 00:00:00 2001
From: Roland Haas <roland.haas@physics.gatech.edu>
Date: Mon, 17 Dec 2012 13:43:02 -0800
Subject: [PATCH] CarpetProlongateTest: restrict tests to 2 processes, reduce
 regions, remove sums, add 0d errornorm ouput instead

required since the tests produce >0D Carpet output which is process
number dependend. Regions sizes generated by CarpetRegrid2 changed since
data was generated, most likely due to
549447bb134500f38da40957c4c52753a1459532 "CarpetRegrid2: Enforce certain
grid hierarchy properties" which unfortunately no longer compiles with
the current flesh.
---
 CarpetExtra/CarpetProlongateTest/test/test.ccl     |   31 ++++++++++++++++++++
 .../CarpetProlongateTest/test/test_cc_rest_o0.par  |    4 +--
 .../carpetprolongatetest::errornorm..asc           |   13 ++++++++
 .../carpetprolongatetest::scaled.sum.asc           |    8 -----
 .../test/test_cc_rest_o0/test_cc_rest_o0.par       |    8 +++--
 .../CarpetProlongateTest/test/test_cc_rest_o1.par  |    4 +--
 .../carpetprolongatetest::errornorm..asc           |   13 ++++++++
 .../carpetprolongatetest::scaled.sum.asc           |    8 -----
 .../test/test_cc_rest_o1/test_cc_rest_o1.par       |    8 +++--
 .../CarpetProlongateTest/test/test_cc_tvd.par      |    2 +-
 .../CarpetProlongateTest/test/test_cc_tvd_hi.par   |    2 +-
 .../CarpetProlongateTest/test/test_cc_tvd_x.par    |    2 +-
 .../CarpetProlongateTest/test/test_cc_tvd_y.par    |    2 +-
 .../CarpetProlongateTest/test/test_cc_tvd_z.par    |    2 +-
 CarpetExtra/CarpetProlongateTest/test/test_o1.par  |    2 +-
 .../test_o1/carpetprolongatetest::scaled.sum.asc   |    8 -----
 16 files changed, 77 insertions(+), 40 deletions(-)
 create mode 100644 CarpetExtra/CarpetProlongateTest/test/test.ccl
 create mode 100644 CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/carpetprolongatetest::errornorm..asc
 delete mode 100644 CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/carpetprolongatetest::scaled.sum.asc
 create mode 100644 CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/carpetprolongatetest::errornorm..asc
 delete mode 100644 CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/carpetprolongatetest::scaled.sum.asc
 delete mode 100644 CarpetExtra/CarpetProlongateTest/test/test_o1/carpetprolongatetest::scaled.sum.asc

diff --git a/CarpetExtra/CarpetProlongateTest/test/test.ccl b/CarpetExtra/CarpetProlongateTest/test/test.ccl
new file mode 100644
index 0000000..64715cb
--- /dev/null
+++ b/CarpetExtra/CarpetProlongateTest/test/test.ccl
@@ -0,0 +1,31 @@
+NPROCS 2
+
+TEST test_interp
+{
+NPROCS 1
+}
+
+TEST test_cc_tvd
+{
+NPROCS 1
+}
+
+TEST test_cc_tvd_hi
+{
+NPROCS 1
+}
+
+TEST test_cc_tvd_x
+{
+NPROCS 1
+}
+
+TEST test_cc_tvd_y
+{
+NPROCS 1
+}
+
+TEST test_cc_tvd_z
+{
+NPROCS 1
+}
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0.par
index 10448ff..c573320 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0.par
@@ -100,7 +100,7 @@ ActiveThorns = "CarpetIOASCII"
 
 IOASCII::one_file_per_group = yes
 
-IOASCII::out0D_every = 0
+IOASCII::out0D_every = 1
 IOASCII::out0D_vars  = "
         CarpetProlongateTest::errornorm
 "
@@ -123,7 +123,7 @@ ActiveThorns = "CarpetIOScalar"
 
 IOScalar::one_file_per_group = yes
 
-IOScalar::outScalar_every      = 1
+IOScalar::outScalar_every      = 0 # disabled
 IOScalar::outScalar_reductions = "sum"
 IOScalar::outScalar_vars       = "
         CarpetProlongateTest::scaled
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/carpetprolongatetest::errornorm..asc b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/carpetprolongatetest::errornorm..asc
new file mode 100644
index 0000000..bf1dc88
--- /dev/null
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/carpetprolongatetest::errornorm..asc
@@ -0,0 +1,13 @@
+# 0D ASCII output created by CarpetIOASCII
+# created on horizon.tapir.caltech.edu by rhaas on Dec 18 2012 at 10:40:16-0800
+# parameter filename: "test_cc_rest_o0.par"
+#
+# CARPETPROLONGATETEST::ERRORNORM (carpetprolongatetest::errornorm)
+#
+# iteration 0   time 0
+# time level 0
+# refinement level 0   multigrid level 0   map 0   component 0
+# column format: 1:it	2:tl	3:rl 4:c 5:ml	6:ix 7:iy 8:iz	9:time	10:x 11:y 12:z	13:data
+# data columns: 13:errornorm
+0	0	0 0 0	0 0 0	0	0 0 0	0
+
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/carpetprolongatetest::scaled.sum.asc b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/carpetprolongatetest::scaled.sum.asc
deleted file mode 100644
index 64458b9..0000000
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/carpetprolongatetest::scaled.sum.asc
+++ /dev/null
@@ -1,8 +0,0 @@
-# Scalar ASCII output created by CarpetIOScalar
-# created on erik-schnetters-macbook-pro.local by eschnett on Feb 11 2011 at 15:28:02-0500
-# parameter filename: "/Users/eschnett/EinsteinToolkit-hg-vanilla/arrangements/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0.par"
-#
-# CARPETPROLONGATETEST::uscaled (carpetprolongatetest::scaled)
-# 1:iteration 2:time 3:data
-# data columns: 3:uscaled
-0 0 0
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/test_cc_rest_o0.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/test_cc_rest_o0.par
index 183f620..c573320 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/test_cc_rest_o0.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o0/test_cc_rest_o0.par
@@ -8,13 +8,15 @@ IO::out_dir = $parfile
 
 
 
+ActiveThorns = "AEILocalInterp"
+
 ActiveThorns = "InitBase"
 
 ActiveThorns = "LoopControl"
 
 
 
-ActiveThorns = "Carpet CarpetLib CarpetReduce"
+ActiveThorns = "Carpet CarpetLib CarpetInterp CarpetReduce"
 
 Carpet::domain_from_coordbase = yes
 Carpet::max_refinement_levels = 10
@@ -98,7 +100,7 @@ ActiveThorns = "CarpetIOASCII"
 
 IOASCII::one_file_per_group = yes
 
-IOASCII::out0D_every = 0
+IOASCII::out0D_every = 1
 IOASCII::out0D_vars  = "
         CarpetProlongateTest::errornorm
 "
@@ -121,7 +123,7 @@ ActiveThorns = "CarpetIOScalar"
 
 IOScalar::one_file_per_group = yes
 
-IOScalar::outScalar_every      = 1
+IOScalar::outScalar_every      = 0 # disabled
 IOScalar::outScalar_reductions = "sum"
 IOScalar::outScalar_vars       = "
         CarpetProlongateTest::scaled
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1.par
index c11cc35..8a4a078 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1.par
@@ -100,7 +100,7 @@ ActiveThorns = "CarpetIOASCII"
 
 IOASCII::one_file_per_group = yes
 
-IOASCII::out0D_every = 0
+IOASCII::out0D_every = 1
 IOASCII::out0D_vars  = "
         CarpetProlongateTest::errornorm
 "
@@ -123,7 +123,7 @@ ActiveThorns = "CarpetIOScalar"
 
 IOScalar::one_file_per_group = yes
 
-IOScalar::outScalar_every      = 1
+IOScalar::outScalar_every      = 0 # disabled
 IOScalar::outScalar_reductions = "sum"
 IOScalar::outScalar_vars       = "
         CarpetProlongateTest::scaled
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/carpetprolongatetest::errornorm..asc b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/carpetprolongatetest::errornorm..asc
new file mode 100644
index 0000000..913d8dd
--- /dev/null
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/carpetprolongatetest::errornorm..asc
@@ -0,0 +1,13 @@
+# 0D ASCII output created by CarpetIOASCII
+# created on horizon.tapir.caltech.edu by rhaas on Dec 18 2012 at 10:40:25-0800
+# parameter filename: "test_cc_rest_o1.par"
+#
+# CARPETPROLONGATETEST::ERRORNORM (carpetprolongatetest::errornorm)
+#
+# iteration 0   time 0
+# time level 0
+# refinement level 0   multigrid level 0   map 0   component 0
+# column format: 1:it	2:tl	3:rl 4:c 5:ml	6:ix 7:iy 8:iz	9:time	10:x 11:y 12:z	13:data
+# data columns: 13:errornorm
+0	0	0 0 0	0 0 0	0	0 0 0	1.4210854715202e-14
+
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/carpetprolongatetest::scaled.sum.asc b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/carpetprolongatetest::scaled.sum.asc
deleted file mode 100644
index 5315052..0000000
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/carpetprolongatetest::scaled.sum.asc
+++ /dev/null
@@ -1,8 +0,0 @@
-# Scalar ASCII output created by CarpetIOScalar
-# created on erik-schnetters-macbook-pro.local by eschnett on Feb 11 2011 at 15:28:08-0500
-# parameter filename: "/Users/eschnett/EinsteinToolkit-hg-vanilla/arrangements/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1.par"
-#
-# CARPETPROLONGATETEST::uscaled (carpetprolongatetest::scaled)
-# 1:iteration 2:time 3:data
-# data columns: 3:uscaled
-0 0 3.5527136788005e-13
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/test_cc_rest_o1.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/test_cc_rest_o1.par
index a16b0af..8a4a078 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/test_cc_rest_o1.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_rest_o1/test_cc_rest_o1.par
@@ -8,13 +8,15 @@ IO::out_dir = $parfile
 
 
 
+ActiveThorns = "AEILocalInterp"
+
 ActiveThorns = "InitBase"
 
 ActiveThorns = "LoopControl"
 
 
 
-ActiveThorns = "Carpet CarpetLib CarpetReduce"
+ActiveThorns = "Carpet CarpetLib CarpetInterp CarpetReduce"
 
 Carpet::domain_from_coordbase = yes
 Carpet::max_refinement_levels = 10
@@ -98,7 +100,7 @@ ActiveThorns = "CarpetIOASCII"
 
 IOASCII::one_file_per_group = yes
 
-IOASCII::out0D_every = 0
+IOASCII::out0D_every = 1
 IOASCII::out0D_vars  = "
         CarpetProlongateTest::errornorm
 "
@@ -121,7 +123,7 @@ ActiveThorns = "CarpetIOScalar"
 
 IOScalar::one_file_per_group = yes
 
-IOScalar::outScalar_every      = 1
+IOScalar::outScalar_every      = 0 # disabled
 IOScalar::outScalar_reductions = "sum"
 IOScalar::outScalar_vars       = "
         CarpetProlongateTest::scaled
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd.par
index 0b07120..f786f1f 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd.par
@@ -83,7 +83,7 @@ CarpetRegrid2::verbose        = yes
 CarpetRegrid2::num_centres = 1
 
 CarpetRegrid2::num_levels_1 = 2
-CarpetRegrid2::radius_1[1]  = 0.2
+CarpetRegrid2::radius_1[1]  = 0.18
 
 
 
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_hi.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_hi.par
index b14cf64..67bd463 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_hi.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_hi.par
@@ -83,7 +83,7 @@ CarpetRegrid2::verbose        = yes
 CarpetRegrid2::num_centres = 1
 
 CarpetRegrid2::num_levels_1 = 2
-CarpetRegrid2::radius_1[1]  = 0.2
+CarpetRegrid2::radius_1[1]  = 0.19
 
 
 
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_x.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_x.par
index 20b64fe..8e51818 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_x.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_x.par
@@ -83,7 +83,7 @@ CarpetRegrid2::verbose        = yes
 CarpetRegrid2::num_centres = 1
 
 CarpetRegrid2::num_levels_1 = 2
-CarpetRegrid2::radius_1[1]  = 0.2
+CarpetRegrid2::radius_1[1]  = 0.18
 
 
 
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_y.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_y.par
index b5969aa..545a210 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_y.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_y.par
@@ -83,7 +83,7 @@ CarpetRegrid2::verbose        = yes
 CarpetRegrid2::num_centres = 1
 
 CarpetRegrid2::num_levels_1 = 2
-CarpetRegrid2::radius_1[1]  = 0.2
+CarpetRegrid2::radius_1[1]  = 0.18
 
 
 
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_z.par b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_z.par
index 89f2305..f084d29 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_z.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_cc_tvd_z.par
@@ -83,7 +83,7 @@ CarpetRegrid2::verbose        = yes
 CarpetRegrid2::num_centres = 1
 
 CarpetRegrid2::num_levels_1 = 2
-CarpetRegrid2::radius_1[1]  = 0.2
+CarpetRegrid2::radius_1[1]  = 0.18
 
 
 
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_o1.par b/CarpetExtra/CarpetProlongateTest/test/test_o1.par
index a3a80ed..c668385 100644
--- a/CarpetExtra/CarpetProlongateTest/test/test_o1.par
+++ b/CarpetExtra/CarpetProlongateTest/test/test_o1.par
@@ -114,7 +114,7 @@ ActiveThorns = "CarpetIOScalar"
 
 IOScalar::one_file_per_group = yes
 
-IOScalar::outScalar_every      = 1
+IOScalar::outScalar_every      = 0 # disabled
 IOScalar::outScalar_reductions = "sum"
 IOScalar::outScalar_vars       = "
         CarpetProlongateTest::scaled
diff --git a/CarpetExtra/CarpetProlongateTest/test/test_o1/carpetprolongatetest::scaled.sum.asc b/CarpetExtra/CarpetProlongateTest/test/test_o1/carpetprolongatetest::scaled.sum.asc
deleted file mode 100644
index 63b5f4d..0000000
--- a/CarpetExtra/CarpetProlongateTest/test/test_o1/carpetprolongatetest::scaled.sum.asc
+++ /dev/null
@@ -1,8 +0,0 @@
-# Scalar ASCII output created by CarpetIOScalar
-# created on erik-schnetters-macbook-pro.local by eschnett on Feb 11 2011 at 15:28:37-0500
-# parameter filename: "/Users/eschnett/EinsteinToolkit-hg-vanilla/arrangements/CarpetExtra/CarpetProlongateTest/test/test_o1.par"
-#
-# CARPETPROLONGATETEST::uscaled (carpetprolongatetest::scaled)
-# 1:iteration 2:time 3:data
-# data columns: 3:uscaled
-0 0 5.6843418860808e-13
-- 
1.7.10.4

